actors/builtin/miner/miner_state_test.go

```diff
@@ -440,6 +442,252 @@ func TestVestingFundsStore(t *testing.T) {
 	assert.Equal(t, abi.NewTokenAmount(51), vested)
 }
 
+// TODO make top level tests for vested, unvested and then together.
+
+func TestVestingFunds(t *testing.T) {
+	t.Run("Vest all funds in single epoch", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   1,
+			StepDuration: 1,
+			Quantization: 1,
+		}
+		vestStart := abi.ChainEpoch(10)
+		vestSum := abi.NewTokenAmount(100)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		amountVested := harness.unlockVestedFunds(vestStart + 2)
+		assert.Equal(t, vestSum, amountVested)
+		assert.Zero(t, harness.s.LockedFunds.Int64())
+		assert.True(t, harness.vestingFundsStoreEmpty())
+		// subsequent calls to unlock return 0
+		amountVested = harness.unlockVestedFunds(vestStart + 2)
+		assert.Zero(t, amountVested)
+
+	})
+	t.Run("Sum of all entries in vesting store is equal to vested sum", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   10,
+			StepDuration: 1,
+			Quantization: 2,
+		}
+		vestStart := abi.ChainEpoch(10)
+		vestSum := abi.NewTokenAmount(100)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		// Entry: 12 Funds: 20
+		// Entry: 14 Funds: 20
+		// Entry: 16 Funds: 20
+		// Entry: 18 Funds: 20
+		// Entry: 20 Funds: 20
+
+		totalVested := abi.NewTokenAmount(0)
+		for e := vestStart; e <= vestStart+vspec.VestPeriod+1; e++ {
+			amountVested := harness.unlockVestedFunds(e)
+			if e == 13 || e == 15 || e == 17 || e == 19 || e == 21 {
```

Comment:
These are pretty gross. Manually calculated numbers for the amount vested are ok, but we can do better here. 

if you use relatively small numbers, I think it would be better to just write out the expected vesting schedules per epoch directly, like I have done in the comment above.